cli/lib/kontena/cli/stacks/list_command.rb

```diff
@@ -37,27 +37,30 @@ def get_stacks
       client.get("grids/#{current_grid}/stacks")['stacks'].tap { |stacks| stacks.map { |stack| stack['depth'] = 0 } }
     end
 
+    # Defines a set of columns for the command.
     def fields
       return ['name'] if quiet?
       {
         name: 'name',
         stack: 'stack',
         services: 'services_count',
         state: 'state',
-        'exposed ports' => 'ports'
+        'exposed ports' => 'ports',
+        labels: 'labels'
       }
     end
 
     def execute
-    stacks = build_depths(get_stacks)
+      stacks = build_depths(get_stacks)
 
-    print_table(stacks) do |row|
+      print_table(stacks) do |row|
         next if quiet?
         row['name'] = health_icon(stack_health(row)) + " " + tree_icon(row) + " " + row['name']
         row['stack'] = "#{row['stack']}:#{row['version']}"
         row['services_count'] = row['services'].size
         row['ports'] = stack_ports(row).join(',')
         row['state'] = pastel.send(state_color(row['state']), row['state'])
+        row['labels'] = stack_labels(row)
```

Comment:
I think this could use some kind of max length thing?

Something like:

```ruby
labels = stack_labels(row)
labels = "#{labels[0..17]}..." if labels.length > 20
row['labels'] = labels.length > 20 ? "#{labels[0..17]}..." : labels
```

Reply:
And maybe only do that if `$stdin.tty?` so that something like `label ls | grep xyz` will not get cut.

Reply:
@kke sounds like a good idea, will adjust for that case as well.

Reply:
✅ RESOLVED